### PR TITLE
Fixes the second param completion

### LINKED ISSUE #9: The package should only ship the product, not dev dependencies
Currently it includes the `linter` and `license_check` packages.
These aren't required in the shipped product.

https://github.com/Azure/azure-cli-shell/blob/master/setup.py#L37

Same with `autopep8` and `flake8 `.

### LINKED ISSUE #10: URL in setup.py is wrong
Currently, it's `https://github.com/oakeyc/azure-cli-interactive-shell`.

### LINKED ISSUE #15: Missing completion on the second parameter
For instance: `vm stop --resourcegroup Test --` does not suggest name

image

See the first completion:
image

### LINKED ISSUE #31: "step by step" if not integer should not show global
Why showing the whole help?

image

### LINKED ISSUE #32: "step by step" if the example does not exists should not print the whole help
I asked for example 2, but there is not example 2. I get the whole basic help, I don't se a reason why:

image

### LINKED ISSUE #37: 'az' shows up again in the combo intellisense after 'az '
Suggest not propose `az` after `az`

screen shot 2017-04-06 at 12 27 18 am

### PR DESCRIPTION
**Pull request by oakeyc**
fixes #15, fixes #10, fixes #37, fixes #9, fixes #31, fixes #32

### CODE REVIEW

File: azclishell/az_completer.py [hunk 1]
```diff
@@ -131,6 +131,7 @@ def get_completions(self, document, complete_event):
 
         text = reformat_cmd(text)
         if text.split():
+
```
Comment (tjprescott) [inline azclishell/az_completer.py:134]: Unless GitHub is being super weird... how does adding a blank line fix all these issues? 😅

**Review by tjprescott — CHANGES_REQUESTED** [review]
Does this actually fix anything?  The only change I see is the addition of a single blank line.

File: azclishell/az_completer.py (hunk 1 above)
Comment (oakeyc) [reply azclishell/az_completer.py:134]: Haha, I don't know why it's doing that, but it does. In commit: https://github.com/Azure/azure-cli-shell/pull/40/commits/41c56d53893f7364c5baac332d6dbf8557d8908c this is the bulk of what I did.

Comment (derekbekoe) [reply azclishell/az_completer.py:134]: Those changes were already merged in in a different PR it seems.

Comment (tjprescott) [reply azclishell/az_completer.py:134]: That's one way of doing it.

**Review by tjprescott — APPROVED** [review]

File: azclishell/az_completer.py (hunk 1 above)
Comment (derekbekoe) [reply azclishell/az_completer.py:134]: However, it doesn't look like it has addressed everything. e.g. https://github.com/Azure/azure-cli-shell/issues/9
@oakeyc please take a look.

**Pull request merged**